pkg/providers/cloudstack/validator.go

```diff
@@ -148,17 +166,32 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, cloudStac
 }
 
 func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfigSpec anywherev1.CloudStackDatacenterConfigSpec, machineConfig *anywherev1.CloudStackMachineConfig) error {
-	domain := datacenterConfigSpec.Domain
-	zone := datacenterConfigSpec.Zone
-	account := datacenterConfigSpec.Account
-	if err := v.cmk.ValidateTemplatePresent(ctx, domain, zone, account, machineConfig.Spec.Template); err != nil {
-		return fmt.Errorf("validating template: %v", err)
+	if len(datacenterConfigSpec.Domain) <= 0 {
+		return fmt.Errorf("CloudStackDatacenterConfig domain is not set or is empty")
+	}
+	domain, errDomain := v.cmk.ValidateDomainPresent(ctx, datacenterConfigSpec.Domain)
+	if errDomain != nil {
+		return fmt.Errorf("error while checking domain %v", errDomain)
+	}
+
+	zones, err := v.cmk.ValidateZonesPresent(ctx, datacenterConfigSpec.Zones)
+	if err != nil {
+		return fmt.Errorf("error while checking zones %v", err)
 	}
```

Comment:
these validations are on the datacenter config, not the machine config. Shouldn't they be handled elsewhere, not here?

Reply:
I am a bit confused how unit testing is the reason why we are adding datacenter validation here. We can always ensure the order of testing all of validate if that is an issue instead of do it this way as it is effectively the same thing.